metricbeat/module/elasticsearch/node_stats/data.go

```diff
@@ -98,13 +165,48 @@ var (
 			"ms":    c.Int("collection_time_in_millis"),
 		},
 	}
+
+	threadPoolSchema = s.Schema{
+		"threads":   c.Int("threads"),
+		"queue":     c.Int("queue"),
+		"active":    c.Int("active"),
+		"rejected":  c.Int("rejected"),
+		"largest":   c.Int("largest"),
+		"completed": c.Int("completed"),
+	}
 )
 
 type nodesStruct struct {
 	ClusterName string                            `json:"cluster_name"`
 	Nodes       map[string]map[string]interface{} `json:"nodes"`
 }
 
+func pipelineMetricsMapping(node map[string]interface{}, metricSetFields common.MapStr, r mb.ReporterV2) {
+	value, ok := node["ingest"].(map[string]interface{})["pipelines"]
```

Comment:
@ruflin Is this multi-level indexing into the map + type assertion, all in one step, safe to do or should I break this up into 3 separate steps (index into "ingest", then the type assertion, then index into "pipelines") with error checking after each step?